StripeFinancialConnections/StripeFinancialConnections/Source/Common/TestModeAutofillBannerView.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  TestModeAutofillBannerView.swift
+//  StripeFinancialConnections
+//
+//  Created by Mat Schmid on 2024-06-20.
+//
+
+@_spi(STP) import StripeCore
+@_spi(STP) import StripeUICore
+import UIKit
+
+class TestModeAutofillBannerView: UIView {
+
+    enum Context {
+        case account
+        case otp
+
+        var buttonLabel: String {
+            switch self {
+            case .account:
+                STPLocalizedString(
+                    "Use test account",
+                    "Label for a button shown to a user who is in test mode, which will give them the option to autofill mock account credentials."
+                )
+            case .otp:
+                STPLocalizedString(
+                    "Use test code",
+                    "Label for a button shown to a user who is in test mode, which will give them the option to autofill a mock one time passcode."
+                )
+            }
+        }
+    }
+
+    private let context: Context
+    private let didTapAutofill: () -> Void
+
+    // MARK: - Subviews
+
+    private lazy var messageLabel: UILabel = {
+        // Create icon as a text attachement.
+        let icon = Image.info
+            .makeImage(template: true)
+            .withTintColor(.attention300)
+        let textAttachment = NSTextAttachment(image: icon)
+
+        textAttachment.bounds = CGRect(
+            x: 0,
+            y: -2,
+            width: icon.size.width,
+            height: icon.size.height
+        )
+
+        let attributedString = NSMutableAttributedString()
+        attributedString.append(NSAttributedString(attachment: textAttachment))
+        attributedString.append(NSAttributedString(string: STPLocalizedString(
+            " You're in test mode.",
```

Comment:
what do you think of putting the space separately from the localized string? (ex. maybe as simple as another `        attributedString.append(NSAttributedString(string: " ")`)

the practical fear I have is that localization teams could forget to put the space there, and then there's no spacing

the less-practical concern is just a separation of concerns: right now we are putting a space in the main text just to get a space (_but_ the space logically should be separate from text - it just "happens" that the text has a space because we added an icon)